File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() -1;
        int area = 0;
        while(left < right){
            int width = right - left;
            int length = min(height[left],height[right]);
            area = max(area, width*length);
            if(height[left] < height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return area;

    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.swift
class Solution {
    func maxArea(_ height: [Int]) -> Int {
        var n = height.count
        var left = 0
        var right = n-1
        var maxArea = 0
        while left < right {
            let h = min(height[left],height[right])
            let w = right - left
            var area = h * w
            maxArea = max(maxArea,area)
            if height[left] < height[right] {
                left += 1
            }
            else {
                right -= 1
            }
        }
        return maxArea
    }
}